src/__tests__/api.test.ts

```diff
@@ -0,0 +1,140 @@
+import axios from 'axios';
+import { describe, beforeEach, it, expect, vi } from 'vitest';
+import { searchBooks } from '../api/api';
+import type { SearchBooksResponse, Book } from '../api/api';
+import type { Mock } from 'vitest';
+
+type MockedAxiosInstance = {
+  get: ReturnType<typeof vi.fn>;
+  interceptors: {
```

Comment:
The app does not use axios interceptors anywhere and this mock can be removed. 
Keeping unused mocks adds unnecessary complexity and can be misleading.